clean-modular-code/python-pep-8.md

```diff
@@ -14,237 +14,279 @@ kernelspec:
-* <a href="https://github.com/hhatto/autopep8" target="_blank">autopep8</a>, another `Python` package that can be used to modify files to the PEP 8 style guide.
+* **Lowercase letter:** `b`
+* **Uppercase letter:** `B`
+* **lowercase:** `this is all lowercase words`
+* **snake case:** when words are separated by underscores: `lower_case_with_underscores`
+* **Uppercase:** All words are all uppercase letters: `UPPERCASE`
+* **Snake case** upper case: `UPPER_CASE_WITH_UNDERSCORES`
+* **CamelCase:** Every word is capitalized so they visually stand out: `CapitalizedWords`. This is sometimes also referred to as CapWords or StudlyCaps.
+  * Note: When using acronyms in CamelCase, capitalize all the letters of the acronym. Thus HTTPServerError is better than HttpServerError.
+* **mixedCase:** (differs from CapitalizedWords by initial lowercase character!)
+* **Capitalized_Words_With_Underscores:** This approach is not recommended. Use one convention and stick with it.
 
-`Python` community members expect that your code will adhere to the PEP 8 standard, and if it does not, they generally will not be shy to tell you that your code is not "Pythonic"!
+:::
 
 +++ {"editable": true, "slideshow": {"slide_type": ""}}
 
-## Additional Resources
+## Best practices for importing libraries
 
-* <a href="https://www.python.org/dev/peps/pep-0008/" target="_blank">The PEP 8 Style Guide</a>
+### Import Python libraries at the top of your code
+
+It’s good practice to import all required libraries at the top of your **Python** script or in the first cell of a **Jupyter Notebook**. This helps anyone understand what packages are needed to run your code. It also follows PEP 8 conventions.
+
+### Organize your imports into groups
+
+PEP 8 recommends organizing imports in the following order:
+
+1. **Standard Library Imports**: These built-in modules come with Python, such as `os` and `glob`. You can find the full list [here](https://docs.python.org/3/library/index.html).
+2. **Third-Party Imports**: Libraries that you install via `pip`, like `numpy` and `pandas`.
+3. **Local Imports**: Code or modules specific to your project.
+
+Here’s an example following PEP 8 conventions:
+
+```python
+import os
+import glob
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+
+from my_module import my_function
+```
 
-* <a href="https://realpython.com/python-pep8/" target="_blank">How To Write Beautiful Python Code with PEP 8</a>
+### Why organize imports?
 
-* <a href="https://www.safaribooksonline.com/library/view/the-hitchhikers-guide/9781491933213/ch04.html" target="_blank">The Hitchhiker's Guide to Python by Tanya Schlusser and Kenneth Reitz</a>
+Organizing your imports this way ensures your code is readable and follows widely accepted practices. Importing libraries at the top also makes it easier to debug and see which dependencies are required to run the code.
```

Comment:
and merge conflicts, but i think basically we should consolidate "why code style" at the beginning because it's mostly the same reasons